config/initializers/rabl_init.rb

```diff
@@ -75,7 +75,7 @@ def call(template)
   # config.cache_engine = Rabl::CacheEngine.new # Defaults to Rails cache
   # config.perform_caching = false
   # config.escape_all_output = false
-  # config.json_engine = nil # Class with #dump class method (defaults JSON)
+  config.json_engine = ActiveSupport::JSON # Class with #dump class method (defaults JSON)
```

Comment:
I'm a bit concerned about this. We may be changing API responses, causing user scripting to break. We know that the date formats will change, but there may be other changes as well as a result of this - the tests did catch one, and there may be others that don't have coverage.

Reply:
I think the test failure has revealed that we're relying on odd behavior of ::JSON encoding:
Here's the object generated by RABL just before encoding (irrespective of json_engine and *without* my change)

```
=> {:description=>nil,
 :override=>true,
 :parameter_type=>"string",
 :hidden_value?=>false,
 :omit=>nil,
 :required=>false,
 :validator_type=>nil,
 :validator_rule=>nil,
 :merge_overrides=>false,
 :merge_default=>false,
 :avoid_duplicates=>false,
 :override_value_order=>"fqdn\norganization,location\nhostgroup",
 :created_at=>Mon, 15 Apr 2019 00:07:47 UTC +00:00,
 :updated_at=>Mon, 15 Apr 2019 00:07:47 UTC +00:00,
 :use_puppet_default=>nil,
 :puppetclass_name=>
  #<Puppetclass:0x000000000dcbd600
   id: 298486374,
   name: "apache",
   created_at: Mon, 15 Apr 2019 00:07:47 UTC +00:00,
   updated_at: Mon, 15 Apr 2019 00:07:47 UTC +00:00>,
 :parameter=>"custom_class_param",
 :id=>1018350795,
 :puppetclass_id=>298486374,
 :override_values_count=>0,
 :default_value=>"abcdef",
 :environments=>[{:id=>334344675, :name=>"production"}],
 :override_values=>[]}
```

Note that puppetclass_name is the whole object, not its name.

Here's the encoding output with ::JSON
```
"{\"description\":null,\"override\":true,\"parameter_type\":\"string\",\"hidden_value?\":false,\"omit\":null,\"required\":false,\"validator_type\":null,\"validator_rule\":null,\"merge_overrides\":false,\"merge_default\":false,\"avoid_duplicates\":false,\"override_value_order\":\"fqdn\\norganization,location\\nhostgroup\",\"created_at\":\"2019-04-15 00:07:47 UTC\",\"updated_at\":\"2019-04-15 00:07:47 UTC\",\"use_puppet_default\":null,\"puppetclass_name\":\"apache\",\"parameter\":\"custom_class_param\",\"id\":1018350795,\"puppetclass_id\":298486374,\"override_values_count\":0,\"default_value\":\"abcdef\",\"environments\":[{\"id\":334344675,\"name\":\"production\"}],\"override_values\":[]}"
```
and with ActiveSupport::JSON
```
"{\"description\":null,\"override\":true,\"parameter_type\":\"string\",\"hidden_value?\":false,\"omit\":null,\"required\":false,\"validator_type\":null,\"validator_rule\":null,\"merge_overrides\":false,\"merge_default\":false,\"avoid_duplicates\":false,\"override_value_order\":\"fqdn\\norganization,location\\nhostgroup\",\"created_at\":\"2019-04-15T00:07:47.426Z\",\"updated_at\":\"2019-04-15T00:07:47.426Z\",\"use_puppet_default\":null,\"puppetclass_name\":{\"id\":298486374,\"name\":\"apache\",\"created_at\":\"2019-04-15T00:07:47.467Z\",\"updated_at\":\"2019-04-15T00:07:47.467Z\"},\"parameter\":\"custom_class_param\",\"id\":1018350795,\"puppetclass_id\":298486374,\"override_values_count\":0,\"default_value\":\"abcdef\",\"environments\":[{\"id\":334344675,\"name\":\"production\"}],\"override_values\":[]}"
```
My opinion is that the failure revealed by the test is due to an improperly declared attribute in the RABL template. If there are other cases like this they should be fixed. However, I don't see any other cases where we're declaring an attribute like `:as => <something>_name` so I'm feeling pretty good about it.

Reply:
i wonder if @jyejare or someone else from QE can run some automation against this branch to check for any other changes in the api? Even changing the date format could possibly cause issues for users, and other changes could be even more difficult to handle.
Is there some way of working around this in katello to convert the date format before using it? I'm a bit reluctant to change API responses if we can avoid it.